src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py

```diff
@@ -70,6 +70,7 @@ def get_vm_size_completion_list(prefix, action, parsed_args, **kwargs):  # pylin
                       completer=get_resource_name_completion_list('Microsoft.Compute/virtualMachines'))
 register_cli_argument('vm disk', 'disk', validator=validate_vm_disk, help='disk name or id',
                       completer=get_resource_name_completion_list('Microsoft.Compute/disks'))
+register_cli_argument('vm disk detach', 'disk_name', options_list=('--name', '-n'), help='The data disk name.')
```

Comment:
Was this not aliased before?  If not, we should annotate this with Breaking Change.